pkg/tekton/tekton.go

```diff
@@ -0,0 +1,128 @@
+package tekton
+
+import (
+	"context"
+	"fmt"
+
+	tektonoperatorv1alpha1 "github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
+	tektonoperatorclientv1alpha1 "github.com/tektoncd/operator/pkg/client/clientset/versioned/typed/operator/v1alpha1"
+	crdclientv1 "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/version"
+)
+
+// ReconcileTekton ensures that Tekton Pipelines has been installed.
+// If Tekton Pipelines has not been installed, ReconcileTekton will create a TektonConfig object
+// so that the Tekton Operator deploys Tekton Pipelines.
+func ReconcileTekton(ctx context.Context,
+	crdClient crdclientv1.ApiextensionsV1Interface,
+	tektonOperatorClient tektonoperatorclientv1alpha1.OperatorV1alpha1Interface) (*tektonoperatorv1alpha1.TektonConfig, bool, error) {
+	pipelinesInstalled, err := IsTektonPipelinesInstalled(ctx, crdClient)
+	if err != nil {
+		return nil, true, err
+	}
+	if pipelinesInstalled {
+		return nil, false, nil
+	}
+	tektonOperatorInstalled, err := IsTektonOperatorInstalled(ctx, crdClient)
+	if err != nil {
+		return nil, true, err
+	}
+	if !tektonOperatorInstalled {
+		return nil, false, fmt.Errorf("tekton operator not installed")
+	}
+	tektonVersion, err := GetTektonOperatorVersion(ctx, crdClient)
+	if err != nil {
+		return nil, true, fmt.Errorf("failed to determine Tekton Operator version: %v", err)
+	}
+	if tektonVersion.Minor() < 49 {
```

Comment:
I was half expecting this to have increased since last September ... have they been keeping the operator version the same as new tekton versions have come out?

Reply:
They have newer versions, but they haven't been published to operatorhub.io. Filed https://github.com/tektoncd/operator/issues/733